### PR TITLE
cli: improved error message when dump table doesn't exist

### LINKED ISSUE #13966: cli: unhelpful error message for cockroach dump prior to GC window
When you use `--as-of` to dump data as of a certain time, if that time is prior to the gc window, you get a generic, unhelpful error:

```
~/src/github.com/cockroachdb/cockroach$ cockroach dump db1 dump_test --dump-mode=data --as-of='2017-03-07 19:55:00'

INSERT INTO dump_test (id, name) VALUES
	(225594758537183233, 'a'),
	(225594758537248769, 'b'),
	(225594758537281537, 'c'),
	(225594758537314305, 'd'),
	(225594758537347073, e'e\n'),
	(225594758537379841, 'f'),
	(225594758537412609, 'g'),
	(225594758537445377, 'h');
~/src/github.com/cockroachdb/cockroach$ cockroach dump db1 dump_test --dump-mode=data --as-of='2017-03-07 19:50:00'
Error: unable to connect or connection lost.

Please check the address and credentials such as certificates (if attempting to
communicate with a secure cluster).

EOF
Failed running "dump"
```

### PR DESCRIPTION
**Pull request by madelynnblue**
Fixes #13966

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13978)

### CODE REVIEW

File: pkg/cli/dump_test.go [hunk 1]
```diff
@@ -552,4 +552,10 @@ INSERT INTO t (i, j) VALUES
 	if dumpAsOf != want1 {
 		t.Fatalf("expected: %s\ngot: %s", want1, dumpAsOf)
 	}
+
+	if out, err := c.RunWithCaptureArgs([]string{"dump", "d", "t", "--as-of", "2000-01-01 00:00:00"}); err != nil {
+		t.Fatal(err)
+	} else if !strings.Contains(string(out), "table d.t does not exist") {
+		t.Fatalf("unexpected output: %", out)
```
Comment (tamird) [inline pkg/cli/dump_test.go]: Missing fmt verb

**Review by tamird — APPROVED** [review]

File: pkg/cli/dump.go [hunk 2]
```diff
@@ -282,6 +282,9 @@ func getMetadataForTable(conn *sqlConn, dbName, tableName string, ts string) (ta
 			AND DATABASE_NAME = $2
 		`, ts), []driver.Value{tableName, dbName})
 	if err != nil {
+		if err == io.EOF {
+			return tableMetadata{}, errors.Errorf("table %s.%s does not exist", dbName, tableName)
```
Comment (jordanlewis) [inline pkg/cli/dump.go]: Heh, now you know why QueryRow might return io.EOF!

**Review by jordanlewis — APPROVED** [review]
LGTM mod the GoVet failure

**Comment by tamird** [comment]
Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13978)*

**Pull request merged**